connection.go

```diff
@@ -49,6 +49,50 @@ type ConnectionTransport interface {
 	Close()
 }
 
+type connTransport struct {
+	uri             *FMPURI
+	l               LogFactory
+	wef             WrapErrorFunc
+	conn            net.Conn
+	transport       Transporter
+	stagedTransport Transporter
+}
+
+var _ ConnectionTransport = (*connTransport)(nil)
```

Comment:
I like the approach, but we have too much overloading of the term `transport` at this point. Can we maybe do a bit of a naming refactor?